src/tests/auth/test_auth_gitlab.py

```diff
@@ -0,0 +1,202 @@
+import copy
+import json
+import os
+import tempfile
+import time
+import unittest
+
+from tornado import escape
+
+from auth.auth_gitlab import GitlabOAuthAuthenticator
+from model import server_conf
+from tests import test_utils
+from utils import file_utils
+from unittest import TestCase
+from unittest.mock import patch, Mock
+
+if __name__ == '__main__':
+    unittest.main()
+
+mock_time = Mock()
+mock_time.return_value = 10000.01
+mock_dump_sessions_to_file = Mock()
+mock_do_update_groups = Mock()
+mock_do_update_user = Mock()
+mock_request_handler = Mock(**{'get_secure_cookie.return_value': "12345".encode()})
+
+
+class TestAuthConfig(TestCase):
+    @patch('time.time', mock_time)
+    @patch('auth.auth_gitlab.GitlabOAuthAuthenticator.dump_sessions_to_file', mock_dump_sessions_to_file)
+    @patch('auth.auth_gitlab.GitlabOAuthAuthenticator.do_update_groups', mock_do_update_groups)
+    def test_gitlab_oauth(self):
```

Comment:
To be honest I think this test method does too many things. Can we split it into multiple tests with separate assertions? 
I think as a rule of thumb, all test methods should call tested code only once and then perform assertions. For every call, there should a new test_method

What do you think?

Reply:
I agree but just selected the shortest way :) I will try to rewrite tests but not sure how much time it can takes